Modify `.card-header-tabs .nav-link.active` colors

shaneparsons opened the pull request:
Set `.card-header-tabs .nav-link.active` background / border colors to better match card.

Fixes #28820.

shaneparsons committed:
Modify `.card-header-tabs .nav-link.active` colors

Set `.card-header-tabs .nav-link.active` background / border colors to better match card.

shaneparsons commented:
For #28820.

ysds commented:
Should also care for hover?

shaneparsons committed:
Use `border-bottom-color` instead of `border-color` for `.card-header-tabs` changes

Use `border-bottom-color` instead of `border-color`, to ensure only the neccessary part of the border changes.

shaneparsons commented:
Since `background-color` isn't actually changed on hover, and the `border-color` changes are fine, I don't think any hover styles are needed.

I did make a change to my pull request though:

Previous:
`border-color: $card-border-color $card-border-color $card-bg;`

New:
`border-bottom-color: $card-bg;`

Now only the bottom border colour changes, to match the card background.

shaneparsons commented:
Here are some screenshots of the difference the pull request makes:

Before:
![Screen Shot 2019-05-27 at 3 21 46 PM](https://user-images.githubusercontent.com/5495493/58436446-62581b80-8093-11e9-88a0-f317b2424efd.png)

After:
![Screen Shot 2019-05-27 at 3 22 37 PM](https://user-images.githubusercontent.com/5495493/58436454-684dfc80-8093-11e9-8c3b-8b9379a2cc55.png)

Note: the first tab is active, the second tab is hovered.

shaneparsons committed:
Remove whitespace from line 107

mdo commented on scss/_card.scss:
```diff
@@ -104,6 +104,11 @@
   border-bottom: 0;
+
+  .nav-link.active {
+    background: $card-bg;
```
Don’t use the shorthand, use `background-color` so it doesn’t override additional `background-` properties.

```suggestion
    background-color: $card-bg;
```

mdo reviewed (commented):
I think I’m in favor of this. Feels like a helpful consideration for folks who may modify the `$body-bg` to be something different. Otherwise though, it’s unnecessary. Thoughts from @twbs/css-review for how to account for that and avoid including CSS that’s not used by default?

shaneparsons committed:
Use `background-color` instead of `background`

Use `background-color` instead of `background` so it doesn't override additional `background-` properties.

Co-Authored-By: Mark Otto <otto@github.com>

shaneparsons commented:
@mdo Indeed. Some frameworks like Laravel modify the `$body-bg` by default too, so it's one less thing that developers have to worry about.

shaneparsons committed:
Merge branch 'master' into patch-1

shaneparsons commented:
There's probably a way to modify this to play nice with [#background-and-color](https://getbootstrap.com/docs/4.3/components/card/#background-and-color) as well, but that's a seperate issue.

ysds reviewed (changes requested):
> Thoughts from @twbs/css-review for how to account for that and avoid including CSS that’s not used by default?

@shaneparsons Could you change as the follows? 😄

```scss
  .nav-link.active {
    background-color: $card-bg;
    border-bottom-color: $card-bg;
  }
```

to

```scss
  @if $nav-tabs-link-active-bg != $card-bg {
    .nav-link.active {
      background-color: $card-bg;
      border-bottom-color: $card-bg;
    }
  }
```

shaneparsons committed:
Wrap `.nav-link.active` changes in a conditional

shaneparsons commented:
> > Thoughts from @twbs/css-review for how to account for that and avoid including CSS that’s not used by default?
> 
> @shaneparsons Could you change as the follows? 😄
> 
> ```scss
>   .nav-link.active {
>     background-color: $card-bg;
>     border-bottom-color: $card-bg;
>   }
> ```
> 
> to
> 
> ```scss
>   @if $nav-tabs-link-active-bg != $card-bg {
>     .nav-link.active {
>       background-color: $card-bg;
>       border-bottom-color: $card-bg;
>     }
>   }
> ```

Done!

See my edit to https://github.com/twbs/bootstrap/pull/28833#issuecomment-496524571 and let me know if the additions are of any interest to you.

ysds commented:
I think it's over-engineering to integrate with the background utilities. It would be better to solve it with custom variables in the future (v6?)

shaneparsons commented:
> I think it's over-engineering to integrate with the background utilities. It would be better to solve it with custom variables in the future (v6?)

Yeah I agree on that one. I don't have a ton of experience with mixins / utilities, so that was just my hacky take on the solution. I more so just intended to get the conversation started.

MartijnCuppens committed:
Merge branch 'master' into patch-1

mdo reviewed (approved):
Thanks, everyone! <3

XhmikosR commented:
I assume we don't want this backported, right?

XhmikosR committed:
Merge branch 'master' into patch-1
